src/pytest_flyte/plugin.py

```diff
@@ -123,20 +149,57 @@ def docker_cleanup():
 
 
 @pytest.fixture(scope="session")
-def flyteclient(docker_ip, docker_services, docker_compose, capsys_suspender):
-    port = docker_services.port_for("backend", 30081)
-    url = f"{docker_ip}:{port}"
-    os.environ["FLYTE_PLATFORM_URL"] = url
-    os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
-
-    def _check():
-        try:
-            docker_compose.execute("exec backend wait-for-flyte.sh")
-            return True
-        except Exception as e:
-            print(e)
-            return False
-
-    with capsys_suspender():
-        docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
-    return friendly.SynchronousFlyteClient(url, insecure=True)
+def flyteclient(request):
+    if not request.config.getoption("--proto-path"):
+        raise ValueError("Serialized Data Proto Path must be set")
+
+    if request.config.getoption("--local") in ["True", "true", True]:
+        docker_ip = request.getfixturevalue("docker_ip")
+        docker_services = request.getfixturevalue("docker_services")
+        docker_compose = request.getfixturevalue("docker_compose")
+        capsys_suspender = request.getfixturevalue("capsys_suspender")
+
+        port = docker_services.port_for("backend", 30081)
+        url = f"{docker_ip}:{port}"
+        os.environ["FLYTE_PLATFORM_URL"] = url
+        os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
+
+        # create flytectl config
+        os.makedirs(os.path.join(str(Path.home()), ".flyte"), exist_ok=True)
+        config = {
+            "admin": {"endpoint": "dns:///" + url, "insecure": True},
+            "logger": {"show-source": True, "level": 1},
+        }
+
+        with open(
+            os.path.join(str(Path.home()), ".flyte", "config.yaml"), "w"
+        ) as yaml_file:
+            yaml.dump(config, yaml_file, default_flow_style=False)
+
+        def _check():
+            try:
+                docker_compose.execute("exec backend wait-for-flyte.sh")
+                return True
+            except Exception as e:
+                print(e)
+                return False
+
+        with capsys_suspender():
+            docker_services.wait_until_responsive(timeout=900, pause=1, check=_check)
+        return friendly.SynchronousFlyteClient(url, insecure=True)
+
+    else:
+        capsys_suspender = request.getfixturevalue("capsys_suspender")
+        if request.config.getoption("--flyte-platform-url"):
+            os.environ["FLYTE_PLATFORM_URL"] = request.config.getoption(
+                "--flyte-platform-url"
+            )
+            os.environ["FLYTE_PLATFORM_INSECURE"] = "true"
```

Comment:
what if my remote cluster doesn't support insecure mode? before assigning default value, we can add a check on env variable.  Also, print the insecure option value in the log.